### PR TITLE
Register: Validate Birthday

### PR DESCRIPTION
**Pull request by tombettany**
KanoComputing/peldins#1876
Perform validation on the birthday as it is being typed in a similar
fashion to the username and password entries. Also restrict the
birthday inputs to 2 characters for the day and month and 4 for the
year.

cc @pazdera @convolu @carolineclark

### CODE REVIEW

File: kano_registration_gui/GetData.py [hunk 1]
```diff
@@ -180,17 +192,11 @@ def get_cached_username_and_birthday(self):
...
-        full = True
-
-        if not self.username_entry.validated:
-            full = False
-
-        if not self.password_entry.validated:
-            full = False
-
-        bday_filled = self.bday_widget.birthday_entries_filled()
+        if (
```
Comment (pazdera) [inline kano_registration_gui/GetData.py]: Is the linebreak here intentional?

Comment (convolu) [inline kano_registration_gui/GetData.py]: When breaking lines after an if, use twice the indentation, to ensure that the block following the ':' is easily distinguished.

```python
if (width == 0 and height == 0 and
        color == 'red' and emphasis == 'strong' or
        highlight > 100):
    raise ValueError("sorry, you lose")
```

**Comment by convolu** [comment]
Other than a minor styling related comment, it looks good to me @tombettany. :-D

**Comment by tombettany** [comment]
@convolu @pazdera I fixes the formatting problem so I will merge now.

**Pull request merged**